src/mantle/testing/concerns/trait-phpunit-upgrade-warning.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * PHPUnit_Upgrade_Warning trait file
+ *
+ * @package Mantle
+ */
+
+namespace Mantle\Testing\Concerns;
+
+use PHPUnit\Runner\Version;
+use PHPUnit\TextUI\Configuration\Registry;
+use PHPUnit\TextUI\Configuration\TestSuite;
+
+use function Mantle\Support\Helpers\collect;
+use function Termwind\render;
+
+/**
+ * Provide a warning to users who are attempting to run PHPUnit 10+ against an
+ * incompatible codebase.
+ *
+ * PHPUnit 10+ requires code bases to use PSR-4 standards for test classes. We
+ * also need to be careful of any possible changes to the PHPUnit classes we're
+ * referencing because they are not covered by the backward compatibility promise
+ * for PHPUnit.
+ *
+ * @mixin \Mantle\Testing\Installation_Manager
+ */
+trait PHPUnit_Upgrade_Warning {
+	/**
+	 * Whether to silence the PHPUnit 10+ warning.
+	 *
+	 * @var bool
+	 */
+	public bool $silence_phpunit_warning = false;
+
+	/**
+	 * Check if the current codebase is running PHPUnit 10 or higher.
+	 *
+	 * @return bool
+	 */
+	protected function is_running_phpunit_10_or_higher(): bool {
+		// Prevent conflicts if the internal class API changes.
+		if ( ! class_exists( Version::class ) || ! method_exists( Version::class, 'id' ) ) {
+			return false;
+		}
+
+		return version_compare( '10.0.0', Version::id(), '<=' );
+	}
+
+	/**
+	 * Warn the user if they're running PHPUnit 10+ against an incompatible test suite.
+	 *
+	 * @param TestSuite $test_suite Test suite configuration.
```

Comment:
Should this docblock have a return ?

Reply:
No. this repo doesn't use unnecessary returns